include/circt/Dialect/Ibis/IbisPasses.td

```diff
@@ -95,4 +95,42 @@ def IbisArgifyBlocks : Pass<"ibis-argify-blocks"> {
   let constructor = "circt::ibis::createArgifyBlocksPass()";
 }
 
+def IbisReblock : Pass<"ibis-reblock", "ibis::MethodOp"> {
+  let summary = "Recreates `ibis.block` operations from a CFG";
+  let description = [{
+    Recreates `ibis.block` operations from a CFG. Any `ibis.block.attributes`
+    operations at the parent operation will be added to the resulting blocks.
+
+    The IR is expected to be in maximal SSA form prior to this pass, given that
+    the pass will only inspect the terminator operation of a block for any
+    values that are generated within the block. Maximum SSA form thus ensures
+    that any value defined within the block is never used outside of the block.
+
+    It is expected that `ibis.call` operations have been isolated into
```

Comment:
Does this imply that async calls will break up basic blocks?

Reply:
I don't expect async calls to do so. Async calls are per-se not really involved in control seeing as it can't backpressure (i presume) and doesn't generate any values to-be-used downstream (that is, explicitly. I assume one can still create patterns where one loops until a given global condition is true, and thus implicitly rely on the side effects of an async call). As such, i expect async calls to be triggered whenever control reaches its parent scope (i.e. a pipeline stage).